use option "beautify.config"

IgorNovozhilov opened the pull request:
use option `"beautify.config"` specified in the user or workspace settings file, to search for a file of settings for the js-beautify

IgorNovozhilov commented:
Made by analogy with:
https://marketplace.visualstudio.com/items?itemName=dbaeumer.jshint

To support the work of a single configuration for projects stored on different devices.

PS: Your module is the only module that does not support this setting, from the following list:
https://marketplace.visualstudio.com/items?itemName=dbaeumer.jshint
https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint
https://marketplace.visualstudio.com/items?itemName=donjayamanne.python

HookyQR commented on options.js:
```diff
@@ -118,10 +118,13 @@ function set_file_editorconfig_opts(file, config) {
 	let base = vscode.workspace.rootPath;
 	let opts = optionsFromVSCode(doc, formattingOptions, type);
-	let configFile;
+	let configFile = vscode.workspace.getConfiguration('beautify').config;
```
Settings the `config`ed beautifyrc path at this point makes it the default. Perhaps setting it before the call to find it in the home directory (or even after) is a better choice.

Having a `.beautifyrc` in the same, or a parent directory of the target file should be the default. This allows different formatting options for different directories within a project.

Putting it here would break this functionality once a config path is set.

HookyQR commented on package.json:
```diff
@@ -35,6 +35,14 @@
+						"null"
+					],
+					"default": null,
+					"description": "A path to file containing the configuration options for js-beautify. If the file exists it overrides any .jsbeautify file"
```
Again, I wouldn't set this as the default. Perhaps target tree first, then this setting.

IgorNovozhilov commented:
I agree, I like the option:
```
1. path tree, up to project root,
2. config,
3. rest of path tree,
4. home.
```

c' `Also, doing it with a config setting required you to shared you VS Code config anyway, so why not share your .jsbeautifyrc?`

So as, VS Code config is a one file, but configuration files for other tools quite a lot. And if you have a lot of projects, the settings are easier to store in a single folder, for example put it on the corporate VCS resource... Big company, lots of people, lots of projects.

And, yes, the VS Code config is stored in the same folder =)

HookyQR commented:
Cool, I prefer that one too, and coding it will be a little more interesting. Good luck. Looking forward to seeing the new version.

IgorNovozhilov commented:
- checked the behavior with subdirectories in the project folder that you described, it works
- corrected the description of the parameters
- checked the work without the project folder, corrected my mistakes =)
